mesonbuild/modules/wayland.py

```diff
@@ -0,0 +1,115 @@
+# Copyright © 2020 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing as T
+from pathlib import PurePath
+
+from . import ExtensionModule
+from . import ModuleReturnValue
+from .. import build
+from .. import mlog
+from ..interpreterbase import permittedKwargs, stringArgs
+from ..mesonlib import MesonException, File, MachineChoice, version_compare
+
+if T.TYPE_CHECKING:
+    from ..interpreter import ModuleState, Interpreter
+
+
+class WaylandModule(ExtensionModule):
+
+    def __init__(self, interpreter: 'Interpreter'):
+        super().__init__(interpreter)
+
+        # wayland-scanner has a pkg-config file, we should use that to find
+        # wayland-scanner, as it's more robust
+        scanner_dep  = interpreter.dependency_impl(
+            'wayland-scanner', 'wayland-scanner', {'required': True, 'native': True})
+        scanner_name = scanner_dep.variable_method([], {'pkgconfig': 'wayland_scanner'})
```

Comment:
I think you should make the pkg-config dependency optional, and fallback to set scanner_name to "wayland-scanner"

Reply:
One reason for this is wayland could be built as subproject I guess? In that case the pkg-config file won't exist yet but wayland project should have called override_find_program().